Userland/Libraries/LibSQL/Database.cpp

```diff
@@ -18,31 +18,67 @@
 namespace SQL {
 
 Database::Database(String name)
-    : m_heap(Heap::construct(name))
+    : m_heap(Heap::construct(move(name)))
     , m_serializer(m_heap)
-    , m_schemas(BTree::construct(m_serializer, SchemaDef::index_def()->to_tuple_descriptor(), m_heap->schemas_root()))
-    , m_tables(BTree::construct(m_serializer, TableDef::index_def()->to_tuple_descriptor(), m_heap->tables_root()))
-    , m_table_columns(BTree::construct(m_serializer, ColumnDef::index_def()->to_tuple_descriptor(), m_heap->table_columns_root()))
 {
+}
+
+ErrorOr<void> Database::open()
+{
+    TRY(m_heap->open());
+    m_schemas = BTree::construct(m_serializer, SchemaDef::index_def()->to_tuple_descriptor(), m_heap->schemas_root());
     m_schemas->on_new_root = [&]() {
         m_heap->set_schemas_root(m_schemas->root());
     };
+
+    m_tables = BTree::construct(m_serializer, TableDef::index_def()->to_tuple_descriptor(), m_heap->tables_root());
     m_tables->on_new_root = [&]() {
         m_heap->set_tables_root(m_tables->root());
     };
+
+    m_table_columns = BTree::construct(m_serializer, ColumnDef::index_def()->to_tuple_descriptor(), m_heap->table_columns_root());
     m_table_columns->on_new_root = [&]() {
         m_heap->set_table_columns_root(m_table_columns->root());
     };
-    auto default_schema = get_schema("default");
-    if (!default_schema) {
-        default_schema = SchemaDef::construct("default");
-        add_schema(*default_schema);
+
+    m_open = true;
+    auto default_schema_or_error = get_schema("default");
+    if (default_schema_or_error.is_error())
+        return default_schema_or_error.error();
+    if (!default_schema_or_error.value()) {
+        auto default_schema = SchemaDef::construct("default");
+        TRY(add_schema(*default_schema));
     }
+    return {};
```

Comment:
Since `Database::open` returns `ErrorOr`, can this be reduced? Something like:
```c++
    auto default_schema = TRY(get_schema("default"));
    if (!default_schema) {
        default_schema = SchemaDef::construct("default");
        TRY(add_schema(*default_schema));
    }
```

Reply:
I didn't realize `TRY()` could return things. I'm watching @awesomekling's new video as we speak so I'm educated now haha.